crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -474,7 +502,44 @@ fn run_2q_unitary_synthesis(
                 let synth_error_from_target = synth_error(py, scoring_info, target);
                 synth_errors_sequence.push((sequence, synth_error_from_target));
             }
-            DecomposerType::XXDecomposer(_) => {
+            DecomposerType::TwoQubitControlledU(_) => {
+                let sequence = synth_su4_sequence(
+                    py,
+                    &unitary,
+                    decomposer,
+                    preferred_dir,
+                    approximation_degree,
+                )?;
+                let scoring_info =
+                    sequence
+                        .gate_sequence
+                        .gates()
+                        .iter()
+                        .map(|(gate, params, qubit_ids)| {
+                            let inst_qubits =
+                                qubit_ids.iter().map(|q| ref_qubits[*q as usize]).collect();
+                            match gate {
+                                Some(gate) => (
+                                    gate.name().to_string(),
+                                    Some(params.iter().map(|p| Param::Float(*p)).collect()),
+                                    inst_qubits,
+                                ),
+                                None => (
+                                    sequence
+                                        .decomp_gate
+                                        .operation
+                                        .standard_gate()
+                                        .name()
+                                        .to_string(),
+                                    Some(params.iter().map(|p| Param::Float(*p)).collect()),
+                                    inst_qubits,
+                                ),
+                            }
+                        });
+                let synth_error_from_target = synth_error(py, scoring_info, target);
+                synth_errors_sequence.push((sequence, synth_error_from_target));
```

Comment:
I should have come up with a nicer solution at the time, but back when we only had 2 decomposer types and the code for the `scoring_info` was slightly different, I was relatively ok with the code duplication. Now that two of the three arms look almost exactly the same, I wonder if we could abstract the code in a smarter way to avoid so much repetition. I don't think this will affect performance so it can also be left to a follow-up.